funnel/views/proposal.py

```diff
@@ -203,13 +204,47 @@ def view(self):
             for link in self.obj.links.replace('\r\n', '\n').split('\n')
             if link
         ]
-
         return {
             'project': self.obj.project,
             'proposal': self.obj,
             'links': links,
+            'subscribed': bool(self.obj.commentset.current_roles.document_subscriber),
         }
 
+    @route('update_subscribe', methods=['POST'])
+    @requires_login
+    def update_subscribe(self):
+        subscribe_form = ProposalSubscribeForm()
+        subscribe_form.form_nonce.data = subscribe_form.form_nonce.default()
+        if subscribe_form.validate_on_submit():
+            if subscribe_form.subscribe.data:
+                self.obj.commentset.add_subscriber(
+                    actor=current_auth.user, user=current_auth.user
+                )
+                db.session.commit()
+                return {
+                    'status': 'ok',
+                    'message': __("Subscribed"),
+                    'form_nonce': subscribe_form.form_nonce.data,
+                }
+            else:
+                self.obj.commentset.remove_subscriber(
+                    actor=current_auth.user, user=current_auth.user
+                )
+                db.session.commit()
```

Comment:
Current discussion in #1025 says this remove action cannot be triggered by the user. It's automatic as a consequence losing the role from other actions. Instead, user gets a mute action.